src/Development/Shake/Internal/Core/Database.hs

```diff
@@ -101,7 +102,9 @@ setMem :: DatabasePoly k v -> Id -> k -> v -> Locked ()
 setMem Database{..} i k v = liftIO $ Ids.insert status i (k,v)
 
 modifyAllMem :: DatabasePoly k v -> (v -> v) -> Locked ()
-modifyAllMem Database{..} f = liftIO $ Ids.forMutate status $ second f
+modifyAllMem Database{..} f = liftIO $ Ids.forMutate status $ \(k,v) ->
```

Comment:
I wonder, does `status $ \(k, v) -> (k, f v)` also leak?

Reply:
It probably does, because tuples are lazy.

Reply:
Well, maybe `second f` blocks inlining, which often is another reason of space leaks.